kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/resolver/ClusterResolutionResult.java

```diff
@@ -6,109 +6,197 @@
 
 /**
  * The result of a deep resolution of the dependencies of a single VirtualKafkaCluster. Describes
  * whether there were any resolution problems like dangling references, or referenced resources
  * that have a ResolvedRefs: False condition.
+ *
+ * @param cluster the cluster resource being recursively resolved
+ * @param proxyResolutionResult the resolution result for the KafkaProxy resource this cluster references
+ * @param filterResolutionResults resolution results for all filters referenced by the cluster, in the same order they are declared in the VirtualKafkaCluster spec
+ * @param serviceResolutionResult the resolution result for service referenced by the cluster
+ * @param ingressResolutionResults the resolution results for ingresses referenced by the cluster, in the same order they are declared in the VirtualKafkaCluster spec
  */
 public record ClusterResolutionResult(VirtualKafkaCluster cluster,
-                                      Set<DanglingReference> danglingReferences,
-                                      Set<LocalRef<?>> resourcesWithResolvedRefsFalse,
-                                      Set<LocalRef<?>> referentsWithStaleStatus) {
-    public ClusterResolutionResult {
-        Objects.requireNonNull(danglingReferences);
+                                      ResolutionResult<KafkaProxy> proxyResolutionResult,
+                                      List<ResolutionResult<KafkaProtocolFilter>> filterResolutionResults,
+                                      ResolutionResult<KafkaService> serviceResolutionResult,
+                                      List<IngressResolutionResult> ingressResolutionResults) {
+
+    /**
+     * All referents and cluster are fully resolved iff
+     * <ol>
+     *     <li>All references can be retrieved from kubernetes (we have no dangling refs)</li>
+     *     <li>No referent has a ResolvedRefs=False condition, which declares that it has unresolved dependencies</li>
+     *     <li>The cluster does not have a ResolvedRefs=False condition, which declares that it has unresolved dependencies</li>
+     * </ol>
+     * @return true if fully resolved
+     */
+    public boolean allReferentsAndClusterFullyResolved() {
+        return allReferentsFullyResolved() && !isClusterResolvedRefsFalse();
     }
 
-    // using wildcards intentionally
+    /**
+     * All referents are fully resolved iff
+     * <ol>
+     *     <li>All references can be retrieved from kubernetes (we have no dangling refs)</li>
+     *     <li>No referent has a ResolvedRefs=False condition, which declares that it has unresolved dependencies</li>
+     * </ol>
+     * @return true if fully resolved
+     */
+    public boolean allReferentsFullyResolved() {
+        return !anyReferentResolvedRefsConditionsFalse() && !anyDanglingReferences();
+    }
+
+    /**
+     * Transforms this result into a ResolvedCluster, it is assumed that the client has checked that this ClusterResolutionResult
```

Comment:
"Transforms this result into a ResolvedCluster"

I think you need to explain somewhere why you need a separate representations of the result from the resolved things.

Reply:
removed `Resolved*` classes